Rss feed

mulicvedad opened the pull request:
please merge this to master @nejra

mulicvedad committed:
succesfully fetched news from feed

-made helper class FeedDownloader in which i put the logic of
dowloading (URLConnection) and parsing (NSXmlParser) news

mulicvedad committed:
error handling and ui error fixing

nejra commented:
> [!] Unable to satisfy the following requirements: 
> `RestKit (~> 0.27)` required by `Podfile`
> Specs satisfying the `RestKit (~> 0.27)` dependency were found, but they required a higher minimum deployment target.

I get this error when installing pods. Please, change Pods deployment target to match deployment target of your project (we agreed on 9.0, so it should be change in Podfile to `platform :ios, '9.0'` as well as in MovieApp project settings).

nejra commented:
Remove autogenerated unimplemented delegate methods from all files (i.e. from AppDelegate remove everything except `application:didFinishLaunchingWithOptions:` you will be needing that for sure). Same goes for unused classes/files, like `ViewController.h/ViewController.m`.

nejra commented on MovieApp/MovieApp.xcodeproj/project.pbxproj:
```diff
@@ -78,6 +107,17 @@
+			);
+			name = Help;
+			sourceTree = "<group>";
+		};
```
Help group should be renamed to Helpers and/or Utilities to keep some naming standard accross the app. See the difference between these too options.

nejra commented on MovieApp/MovieApp/NewFeedsItem.h:
```diff
@@ -0,0 +1,20 @@
+@property (strong, nonatomic) NSString *headline;
+@property (strong, nonatomic) NSString *text;
+@property (strong, nonatomic) NSString *webPage;
+//@property (strong, nonatomic) NSDate *dateOfArticle;
```
Remove commented code before submitting PR.

nejra commented on MovieApp/MovieApp/NewFeedsItem.h:
```diff
@@ -0,0 +1,20 @@
+@property (strong, nonatomic) NSString *webPage;
+//@property (strong, nonatomic) NSDate *dateOfArticle;
+
+- (id)initWithHeadline:(NSString *)pheadline andWithText:(NSString *)ptext andWithWebPage:(NSString *)pwebPage;
```
Too much unnecessary repetition in a name of the method as well as unclear argument names: `initWithHeadline:andWithText:andWithWebPage:`. You can rename it to something like: `initWithHeadline:text:sourceUrlPath:`.

nejra commented on MovieApp/MovieApp/NewsFeedTableViewController.m:
```diff
@@ -0,0 +1,121 @@
+    
+    feedCell.labelHeadline.text = ((NewFeedsItem *)news[indexPath.row]).headline;
+    feedCell.textViewText.text = ((NewFeedsItem *)news[indexPath.row]).text;
+    feedCell.textViewWebPage.text = ((NewFeedsItem *)news[indexPath.row]).webPage;
```
Value feedCell.textViewWebPage.text should be hyperlink that says "See more details" which will open source in web browser (or web view if we wish not to leave the app). It doesn't make sense to show link to user.

nejra commented on MovieApp/MovieApp/NewFeedsItem.h:
```diff
@@ -0,0 +1,20 @@
+
+@property (strong, nonatomic) NSString *headline;
+@property (strong, nonatomic) NSString *text;
+@property (strong, nonatomic) NSString *webPage;
```
Maybe you can rename this to `sourceUrlPath` or something.

nejra commented on MovieApp/NewsFeedItemView.xib:
```diff
@@ -0,0 +1,20 @@
+            <point key="canvasLocation" x="25.5" y="-222.5"/>
+        </view>
+    </objects>
+</document>
```
I suppose this XIB (along with .h and .m files) should be deleted, since it's unused and not referenced in the project.

nejra commented on MovieApp/MovieApp/CustomTableViewCell.h:
```diff
@@ -0,0 +1,19 @@
+
+- (CGFloat)getHeight;
++(CGFloat)cellHeight;
++(NSString*)cellIdentifier;
```
Remove `getHeight` method, since it is unused. Implement `cellHeight` and `cellIdentifier` methods by setting default values (that can be overridden).

nejra commented on MovieApp/MovieApp/FeedTableViewCell.h:
```diff
@@ -0,0 +1,19 @@
+@interface FeedTableViewCell : CustomTableViewCell
+
+@property (weak, nonatomic) IBOutlet UILabel *labelHeadline;
+@property (weak, nonatomic) IBOutlet UITextView *textViewText;
```
This can be replaced with `UILabel` named `textLabel`. We will display first 5 lines and by finishing paragraph with ellipsis we will indicate that there is more to read (on the web by clicking source link).

nejra commented on MovieApp/MovieApp/FeedTableViewCell.h:
```diff
@@ -0,0 +1,19 @@
+
+@interface FeedTableViewCell : CustomTableViewCell
+
+@property (weak, nonatomic) IBOutlet UILabel *labelHeadline;
```
I prefer calling properties `headlineLabel`, with type indicator as suffix, instead of prefix.

nejra commented on MovieApp/MovieApp/NewsFeedTableViewController.m:
```diff
@@ -0,0 +1,121 @@
+    }
+    
+    feedCell.labelHeadline.text = ((NewFeedsItem *)news[indexPath.row]).headline;
+    feedCell.textViewText.text = ((NewFeedsItem *)news[indexPath.row]).text;
```
Use attributed string to present links in text view (see: http://stackoverflow.com/questions/33121072/add-hyperlinks-to-nsstring-ios).

nejra commented on MovieApp/MovieApp/NewsFeedTableViewController.m:
```diff
@@ -0,0 +1,121 @@
+       
+        feedCell = [tableView dequeueReusableCellWithIdentifier:@"feedCell"];
+
+    }
```
You can remove 

```
if(!feedCell)
 {
        feedCell = [tableView dequeueReusableCellWithIdentifier:@"feedCell"];
 }
```

since it will never be called. you should always dequeue specific cell, that is registered in `viewDidLoad`.

nejra commented on MovieApp/MovieApp/NewsFeedTableViewController.m:
```diff
@@ -0,0 +1,121 @@
+    
+    feedCell.labelHeadline.text = ((NewFeedsItem *)news[indexPath.row]).headline;
+    feedCell.textViewText.text = ((NewFeedsItem *)news[indexPath.row]).text;
+    feedCell.textViewWebPage.text = ((NewFeedsItem *)news[indexPath.row]).webPage;
```
Instead of setting explicitly values to custom cells, you can create method (i.e. `setupWithHeadline:text:sourceUrlPath`)  for doing this (less lines of code and more encapsulated behavior).

mulicvedad replied:
where should i put this method? from the way you wrote its declaration i assume it should be in FeedTableViewCell class

nejra replied:
yes, of course it should be there :)

nejra commented on MovieApp/MovieApp/BoxOfficeFeedDownloader.m:
```diff
@@ -0,0 +1,13 @@
+
+@implementation BoxOfficeFeedDownloader
+
+@end
```
Unused class, please remove it.

mulicvedad committed:
correction of mistakes mentioned in comments

nejra commented on MovieApp/MovieApp/NewsFeedTableViewController.m:
```diff
@@ -0,0 +1,105 @@
++(NSURL *)getFeedsSource
+{
+    return [NSURL URLWithString:@"http://www.boxofficemojo.com/data/rss.php?file=topstories.xml"];
+}
```
Please, move this method to separate configuration file. Same goes for all other configuration stuff you'll have.

nejra commented on MovieApp/MovieApp/NewsFeedTableViewController.m:
```diff
@@ -0,0 +1,105 @@
+    self.navigationItem.leftBarButtonItem = leftButton;
+    self.navigationItem.rightBarButtonItem = rightButton;
+   
+     [self.tableView registerNib:[UINib nibWithNibName:@"FeedTableViewCell" bundle:nil] forCellReuseIdentifier:[FeedTableViewCell cellIdentifier]];
```
Use FeedTableViewCell .cellIdentifier for this, never hardcode it.

nejra commented on MovieApp/MovieApp/NewsFeedTableViewController.m:
```diff
@@ -0,0 +1,105 @@
+    } @catch (NSException *exception) {
+        [news removeAllObjects];
+        [news addObject:[[NewFeedsItem alloc] initWithHeadline:@"Error" text:[exception description] sourceUrlPath:@""]];
+    } 
```
Error handling should be done differently. You can present alert view in this case with descriptive message.

nejra commented on MovieApp/MovieApp/FeedDownloader.h:
```diff
@@ -0,0 +1,25 @@
+
+@interface FeedDownloader : NSObject <NSURLConnectionDelegate, NSXMLParserDelegate>
+
+@property (strong, nonatomic) NSURL * feedURL;
```
Property `feedURL` should not be accessed directly, since it's used only by `FeedDownloader` class. Keep in mind what properties you will expose and if they should be readonly.

nejra commented on MovieApp/MovieApp/FeedDownloader.m:
```diff
@@ -0,0 +1,128 @@
+    _feedURL=feedUrl;
+    
+    NSURLRequest *request = [NSURLRequest requestWithURL:feedUrl];
+    NSURLConnection *conn = [[NSURLConnection alloc] initWithRequest:request delegate:self];
```
Compiler says that `initWithRequest:delegate:` method is depricated since 9.0. Please, change this call as suggested.

nejra commented on MovieApp/MovieApp/NewsFeedTableViewController.m:
```diff
@@ -0,0 +1,105 @@
+        {
+            news = [[NSMutableArray alloc] init];
+        }
+        [news addObject:[[NewFeedsItem alloc] initWithHeadline:@"No results" text:@"" sourceUrlPath:@""]];
```
Or you can have special view for no results case in order to make it more customizable.

mulicvedad committed:
correction of mistakes that @nejra mentioned

note: instead of making special view for ‘no result’ i made alert view
that just says ‘no result’ with ok button. i will make special view
tomorrow

mulicvedad committed:
Merge branch 'master' into rss-feed

mulicvedad committed:
Delete contents.xcworkspacedata

mulicvedad committed:
gitignore changes

mulicvedad committed:
deleted pods folder

mulicvedad committed:
gitignore changed

mulicvedad committed:
flags added

mulicvedad committed:
little corrections that were discussed on skype
